app/Http/Controllers/Api/InventoryController1.php

```diff
@@ -0,0 +1,51 @@
+<?php namespace App\Http\Controllers;
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Item;
+
+use Illuminate\Support\Facades\Response;
+use Illuminate\Support\Facades\Input;
+
+class InventoryController1 extends Controller {
+
+	/**
+	 * Display a listing of the resource.
+	 *
+	 * @param  string $query
+	 * @return json all or queried $items
+	 */
+	public function index($query)
+	{
+		$items = Item::all();
+
+		if(!is_null($query)) {
+			$items = $items->find($query);
+		}
+
+		if(is_null($items)) {
+			return Response::json("Not Found", 404);
+		}
+		return $items;
+	}
+
+
+	/**
+	 * Store a newly created resource in storage.
+	 *
+	 * @param  none
+	 * @return Json input $item
+	 */
+	public function store()
+	{
+		$input = Input::all();
+		if(is_null($input)) {
+			return Response::json("No Input", 406);
+		}
+		Item::create($input);
```

Comment:
I made some changes according to your comments. But I am not sure if they
are correct. Please take a look.

What confuses me is that for the return value of "store", you first said to
return "201" and then you said to return the created item. I chose to
return "201".

Also, I am not sure what to validate in the "store" function, so I just
chose a few that I think may be important. Not sure if they are correct.
Please take a look. Thanks


On Thu, Apr 2, 2015 at 6:35 PM, Zachary May <notifications@github.com>
wrote:

> In app/Http/Controllers/Api/InventoryController1.php
> <https://github.com/zachmay/inventory616/pull/15#discussion_r27703727>:
>
> > +	}
> > +
> > +
> > +	/**
> > +	 * Store a newly created resource in storage.
> > +	 *
> > +	 * @param  none
> > +	 * @return Json input $item
> > +	 */
> > +	public function store()
> > +	{
> > +		$input = Input::all();
> > +		if(is_null($input)) {
> > +			return Response::json("No Input", 406);
> > +		}
> > +		Item::create($input);
>
> Instead of directly returning $input, store the result of calling
> Item::create(...) and return that. That way if there are default values
> for some database columns, they'll be populated in the thing we send back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zachmay/inventory616/pull/15/files#r27703727>.
>